components/private_channel/rust/ffi/src/lib.rs

```diff
@@ -115,8 +121,8 @@ pub unsafe extern "C" fn client_start_challenge(
     std::mem::forget(enc_hashes_buff);
```

Comment:
Better to use `Box::into_raw` than `Box::as_mut_ptr` followed by `mem::forget`.

Reply:
@antonok-edm I changed the logic here and use `ManuallyDrop()` instead. WDYT?